fs/memoryfs.py

```diff
@@ -340,6 +379,11 @@ def _get_dir_entry(self, dir_path):
                 current_entry = current_entry.get_entry(path_component)
             return current_entry
 
+    def close(self):
+        # type: () -> None
+        self.root.clean_up(force=True)
```

Comment:
This may be a simpler operation that it appears. If there are no references to `root` then the memory will be garbage collected, and the `clean_up` method won't be required.

Can you try replacing the clean_up with the following:

```
self.root = None
```

or 

```
del self.root
```

Reply:
Oh yeah, reference counting... :man_facepalming: